src/Cake.Common/Xml/XmlPeekAliases.cs

```diff
@@ -0,0 +1,113 @@
+        /// <param name="xpath">The xpath of the nodes to set.</param>
+        /// <param name="settings">Additional settings to tweak Xml Poke behavior.</param>
+        [CakeMethodAlias]
+        public static string XmlPeek(this ICakeContext context, FilePath filePath, string xpath, XmlPokeSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+            
+            if (filePath == null)
+            {
+                throw new ArgumentNullException("filePath");
+            }
+            
+            var file = context.FileSystem.GetFile(filePath);
+            if (!file.Exists)
+            {
+                throw new FileNotFoundException("Source File not found.", file.Path.FullPath);
+            }
+            
+            using (var memoryStream = new MemoryStream())
+            {
+                using (var fileStream = file.Open(FileMode.Open, FileAccess.Read, FileShare.None))
+                using (var xmlReader = XmlReader.Create(fileStream))
+                using (var xmlWriter = XmlWriter.Create(memoryStream))
+                {
+                    return XmlPeek(xmlReader, xpath, settings);
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Gets the value of a target node.
+        /// </summary>
+        /// <returns>The value found at the given XPath query (or the first, if multiple eligible nodes are found).</returns>
+        /// <param name="source">The source xml to transform.</param>
+        /// <param name="xpath">The xpath of the nodes to set.</param>
+        /// <param name="settings">Additional settings to tweak Xml Poke behavior.</param>
+        private static string XmlPeek(XmlReader source, string xpath, XmlPokeSettings settings)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException("source");
+            }
+            
+            if (string.IsNullOrWhiteSpace(xpath))
+            {
+                throw new ArgumentNullException("xpath");
+            }
+            
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+            
+            var document = new XmlDocument();
+            document.PreserveWhitespace = settings.PreserveWhitespace;
+            document.Load(source);
+            
+            var namespaceManager = new XmlNamespaceManager(document.NameTable);
+            foreach (var xmlNamespace in settings.Namespaces)
+            {
+                namespaceManager.AddNamespace(xmlNamespace.Key /* Prefix */, xmlNamespace.Value /* URI */);
+            }
+            
+            var node = document.SelectSingleNode(xpath, namespaceManager);
+            if (node == null)
+            {
+                const string format = "Failed to find node matching the XPath '{0}'";
```

Comment:
Since we're only querying for a value, wouldn't it be better to log a warning (Failed to find) and return null? This case the consumer don't have to wrap this in try/catch clause.